lib/spack/spack/solver/asp.py

```diff
@@ -1841,11 +1849,23 @@ def setup(self, driver, specs, reuse=None):
         # rules to generate an ASP program.
         self.gen = driver
 
+        # Calculate develop specs
+        # they will be used in addition to command line specs
+        # in determining known versions/targets/os
+        env = ev.active_environment()
+        dev_specs = tuple(
+            spack.spec.Spec(info['spec']).constrained(
+                'dev_path=%s' % info['path']
+            )
+            for name, info in env.dev_specs.items()
+        ) if env else tuple()
+        specs = tuple(specs)  # ensure compatible types to add
```

Comment:
There is some context in which these come in as tuples and come in which they come in as lists, converting to tuples ensures we're able to add the two types together and in this context an immutable type makes sense.

Reply:
oh I think the comment above was meant for here.